src/NLog/Targets/FileTarget.cs

```diff
@@ -1023,18 +1031,15 @@ private void DateAndSequentialArchive(string fileName, string pattern, LogEventI
                         continue;
                     }
 
-                    string datePart = archiveFileName.Substring(fileNameMask.LastIndexOf('*'), dateFormat.Length);
-                    string numberPart = archiveFileName.Substring(fileNameMask.LastIndexOf('*') + dateFormat.Length + 1,
-                        archiveFileName.Length - dateTrailerLength - (fileNameMask.LastIndexOf('*') + dateFormat.Length + 1));
+                    int placeHolderIndex = fileNameMask.LastIndexOf('*');
+                    string dateAndSequence = archiveFileName.Substring(placeHolderIndex, archiveFileName.Length - trailerLength - placeHolderIndex);
+                    string numberPart = dateAndSequence.Substring(dateAndSequence.Length - numDigits);
+                    string datePart = dateAndSequence.Substring(0, dateAndSequence.Length - numberPart.Length - 1);
 
                     int num;
-
-                    try
-                    {
-                        num = Convert.ToInt32(numberPart, CultureInfo.InvariantCulture);
-                    }
-                    catch (FormatException)
+                    if (!Int32.TryParse(numberPart, NumberStyles.None, CultureInfo.InvariantCulture, out num))
```

Comment:
can you add some comments of what we're trying to archive over here? (same for `DateTime.TryParseExact` below). 

Before fully understanding this code, I think it should be a separate method (`detectArchiveFile` or something?) - correct me if wrong.